src/MurmurClient.ts

```diff
@@ -0,0 +1,210 @@
+    extrinsic: Extrinsic,
+    callback: (result: any) => Promise<void> = async () => {}
+  ): Promise<void> {
+    const request: ExecuteRequest = {
+      runtime_call: this.encodeExtrinsic(extrinsic),
+      current_block: (await this.getCurrentBlock()).toNumber(),
+    };
+    try {
+      const response = (await this.http.post("/execute", request))
+        .data as ExecuteResponse;
+
+      const outerExtrinsic = this.idn.tx.murmur.proxy(
+        response.payload.call_data.name,
+        response.payload.call_data.pos,
+        response.payload.call_data.commitment,
+        response.payload.call_data.ciphertext,
+        response.payload.call_data.proof_items,
+        response.payload.call_data.size,
+        extrinsic,
+      )
+
+      this.submitExtrinsic(outerExtrinsic, callback);
+
+      return Promise.resolve();
+    } catch (error) {
+      throw new Error(`Execute failed: ${error}`);
+    }
+  }
+
+  private async getRoundPublic(): Promise<String> {
+    await this.idn.isReady;
+    let roundPublic = await this.idn.query.etf.roundPublic();
+    return roundPublic.toString();
+  }
+
+  private async getCurrentBlock(): Promise<BlockNumber> {
+    await this.idn.isReady;
+    const { number } = await this.idn.rpc.chain.getHeader();
+    return number.unwrap();
+  }
+
+  private defaultMasterAccount(): KeyringPair {
+    const keyring = new Keyring({ type: "sr25519" });
+    const alice = keyring.addFromUri("//Alice");
+    return alice;
+  }
+
+  private async submitExtrinsic(
+    extrinsic: Extrinsic,
+    callback: (result: any) => Promise<void> = async () => {}
+  ): Promise<void> {
+    const unsub = await extrinsic.signAndSend(
+      this.masterAccount,
+      (result: any) => {
+        if (result.status.isInBlock) {
+          console.log(
+            `Transaction included at blockHash ${result.status.asInBlock}`
+          );
+        } else if (result.status.isFinalized) {
+          console.log(
+            `Transaction finalized at blockHash ${result.status.asFinalized}`
+          );
+          unsub();
+          callback(result);
+        }
+      }
+    );
+    return Promise.resolve();
+  }
+
+  private constructExtrinsic(payload: Payload): Extrinsic {
+    let extrinsicPath =
+      `this.idn.tx.${payload.pallet_name}.${payload.call_name}`.toLocaleLowerCase();
+    let parametersPath = "(";
+
+    for (const key in payload.call_data) {
+      if (Array.isArray(payload.call_data[key])) {
+        parametersPath += `[${payload.call_data[key]}], `;
+      } else if (
+        isNaN(payload.call_data[key]) &&
+        payload.call_data[key] != "true" &&
+        payload.call_data[key] != "false"
+      ) {
+        parametersPath += `"${payload.call_data[key]}", `;
+      } else {
+        parametersPath += `${payload.call_data[key]}, `;
+      }
+    }
+
+    parametersPath = parametersPath.slice(0, -2);
+    parametersPath += ")";
+    extrinsicPath += parametersPath;
+    return eval(extrinsicPath);
+  }
```

Comment:
This function is terrible, and it will be removed as part of https://github.com/ideal-lab5/murmur/issues/17